src/CreateEnvelopes.cs

```diff
@@ -346,17 +352,26 @@ private static Profile CreateProfileFromSetbacks(Polygon perimeter, List<Setback
             // perimeter, we have to revert to a boolean strategy.
             foreach (var rs in remainingSetbacks)
             {
-                var a = rs.Baseline.Offset(rs.Distance, false);
-                var b = rs.Baseline.Offset(rs.Distance, true);
-                var rectangleFromSetback = new Polygon(a.Start, a.End, b.End, b.Start);
-                try
+                if (rs.Distance.ApproximatelyEquals(0))
                 {
-                    var difference = Profile.Difference(new[] { profile }, new[] { new Profile(rectangleFromSetback) });
-                    profile = difference.OrderBy(p => Math.Abs(p.Area())).LastOrDefault() ?? profile;
+                    var extented = rs.Baseline.ExtendTo(profile);
+                    var splits = Profile.Split(new[] { profile }, new Polyline(extented.Start, extented.End));
+                    profile = splits.OrderBy(p => Math.Abs(p.Area())).LastOrDefault() ?? profile;
```

Comment:
I agree with your concern about this, but I think this is fine for now — it is rare we will get here as setbacks should generally be drawn along the site perimeter. Let's leave a comment here (and below) like:
```
// This area strategy will choose the wrong polygon in some circumstances. 
// TODO: analyze which side of the setback the results are on, and only include results "inside" of the setback line.
```